src/test/java/cloud/eppo/ApiEndpointsTest.java

```diff
@@ -0,0 +1,77 @@
+package cloud.eppo;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+public class ApiEndpointsTest {
+  final SdkKey plainKey = new SdkKey("flat token");
+
+  @Test
+  public void testDefaultBaseUrl() {
+    ApiEndpoints endpoints = new ApiEndpoints(plainKey, null);
+    assertEquals("https://fscdn.eppo.cloud/api", endpoints.getBaseUrl());
+  }
+
+  @Test
+  public void testCustomBaseUrl() {
+    ApiEndpoints endpoints = new ApiEndpoints(plainKey, "custom.domain/api");
+    assertEquals("https://custom.domain/api", endpoints.getBaseUrl());
+  }
+
+  @Test
+  public void testCustomBaseUrlWithProtocol() {
```

Comment:
Should we have a test that includes a port?

Reply:
yes def